elytron/src/main/java/org/wildfly/extension/elytron/JdbcRealmDefinition.java

```diff
@@ -653,6 +661,28 @@ private AttributeMapper[] resolveAttributeMappers(OperationContext context, Mode
         }
     }
 
+    private static class JdbcRealmDefinitionValidation implements OperationStepHandler {
+
+        @Override
+        public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+            List<ModelNode> principalQueries = PrincipalQueryAttributes.PRINCIPAL_QUERIES_7_0.resolveModelAttribute(context, operation).asList();
```

Comment:
@pedro-hos Unfortunately 'resolveModelAttribute' shouldn't be called in Stage.MODEL. Expressions meant to be resolved from secure sources (i.e. Elytron credential stores) can't be resolved in MODEL. We don't ban their use in MODEL (i.e. throw an exception) because there are a couple valid cases, but this isn't one. (The odds of such an expression being used here are probably minuscule, but we shouldn't rely on that.)

I think the thing to do is:

ModelNode model = context.readResource(PathAddress.EMPTY_ADDRESS, false).getModel();
ModelNode queries = model.get(PrincipalQueryAttributes.PRINCIPAL_QUERIES_7_0.getName());

And then inspect 'queries'.

The 'ModelNode model = context.readResource(PathAddress.EMPTY_ADDRESS, false).getModel();' is because by the time this runs the 'operation' contents have already been processed and stored into the model. It is better to use the results of that work (which may have somehow transformed things) that to rely on the original operation.

Reply:
Sure, I have made the changes. Thank you for the feedback